File: python_and_cpp_practice/job_schedueling.cpp
#include <iostream>
#include <algorithm>
#include<ranges>
using namespace std;
#include <cmath>

int job(int size, int deadline[],int profit[]){
    // int maxdeadline=max_(deadline, deadline + size);
    int maxdeadline = *max_element(deadline, deadline + size);
    int arr[size];
    zip(deadline,profit);
    sort(deadline, deadline+size);
    // for(int i=0;i<size;i++){
    //     cout<<deadline[i]<<" ";
    // }
    sort(profit, profit + size);


}

int main()
{
    int size=6;
    int deadline[size] = {2,3,1,1,2,1};
    int profit[size]={100,20,35,27,30,28};
    job(size, deadline, profit);
}

File: DAA Labs and Lectures/Lab7/job schedueling.py
def job_scheduling(deadline, profit):
    n = len(deadline)
    jobs = list(zip(profit, deadline))
    jobs.sort(reverse=True)
    max_deadline = max(deadline)
    slots = [-1] * (max_deadline + 1)
    total_profit = 0
    scheduled_jobs = []

    for p, d in jobs:
        for slot in range(d, 0, -1):
            if slots[slot] == -1:
                slots[slot] = p
                total_profit += p
                scheduled_jobs.append((p, d))
                break

    return scheduled_jobs, total_profit



deadline = [2, 3, 1, 1, 2, 1]
profit   = [100, 20, 35, 27, 30, 28]
scheduled, profit_sum = job_scheduling(deadline, profit)
print("Scheduled Jobs (profit, deadline):", scheduled)
print("Total Maximum Profit:", profit_sum)
